backend/tests/unit/pidstore/test_providers/test_bai_provider.py

```diff
@@ -0,0 +1,85 @@
+# -*- coding: utf-8 -*-
+#
+# Copyright (C) 2020 CERN.
+#
+# inspirehep is free software; you can redistribute it and/or modify it under
+# the terms of the MIT License; see LICENSE file for more details.
+import mock
+import pytest
+from invenio_pidstore.models import PersistentIdentifier, PIDStatus
+from sqlalchemy.exc import IntegrityError
+
+from inspirehep.pidstore.errors import PIDAlreadyExistsError
+from inspirehep.pidstore.providers.bai import InspireBAIProvider
+
+
+def test_bai_create_fails_after_retrying_when_bai_changes():
```

Comment:
instead of unit tests we could create an integration test with existing bai and mock the provider with the same value, on the first time, and see if it raises the proper errors etc.

Reply:
Why? What's the difference? This will test exactly same code in the same way, but additionally it will add another ~5s to integration tests (setup before and after test, cleaning ES, adding entry to db). I don't see the point of doing this.

Reply:
we're mocking so many things from the app to providers, if someone for example changes the provider significantly these tests will continue to pass. 

With an integration we'll test end-to-end the functionality and in some cases (i.e. the test with `PIDAlreadyExistsError`) we will not even need to mock anything. This is a core feature and I believe we should test

Reply:
But we already have a lot of tests for BAI minter/providers https://github.com/inspirehep/inspirehep/blob/e3d4a17b6133d967fb4d0660243e42cfa5958a94/backend/tests/integration/pidstore/minters/test_minters_bai.py#L99-L265
here I'm testing only the retry.

Reply:
it is not end-to-end tested, you could integrate some of these assertions already in the other tests and just add one that checks the retry functionality.